### PR TITLE
CMake: wxUSE_NO_RTTI from configure has been added

### PR DESCRIPTION
**Pull request by Deamhan**
Just one more option from configure has been added to CMake project. When existing CMake related PRs will be merged I will add bunch of such options.

### CODE REVIEW

File: build/cmake/init.cmake [hunk 1]
```diff
@@ -50,6 +50,15 @@ if(wxBUILD_COMPATIBILITY VERSION_LESS 3.1)
     set(WXWIN_COMPATIBILITY_3_0 ON)
 endif()
 
+if(wxUSE_NO_RTTI)
+    if(MSVC)
+        set(RTTI_COMPILER_FLAGS "/GR-")
+    elseif(CMAKE_COMPILER_IS_GNUCC)
```
Comment (MaartenBent) [inline build/cmake/init.cmake]: It is [suggested](https://cmake.org/cmake/help/latest/variable/CMAKE_COMPILER_IS_GNUCC.html) to use `"${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU"`. 
Also relevant for your other PRs.

**Review by MaartenBent — COMMENTED** [review]
Thanks for these CMake improvements.

Are you only targeting GCC? Most PRs will probably work with Clang as well, I'll test this on Windows and Linux once all of them get merged.

File: build/cmake/init.cmake [hunk 2]
```diff
@@ -50,6 +50,15 @@ if(wxBUILD_COMPATIBILITY VERSION_LESS 3.1)
...
+        set(RTTI_COMPILER_FLAGS "-fno-rtti")
+    endif()
+    wx_string_append(CMAKE_CXX_FLAGS " -DwxNO_RTTI ${RTTI_COMPILER_FLAGS}")
```
[the remark this answers is not in the document]
Comment (Deamhan) [reply build/cmake/init.cmake]: Yes, I know. I did as I did just for the sake of a common style. I will gladly change it, thank you.

File: build/cmake/init.cmake (hunk 1 above)
Comment (Deamhan) [reply build/cmake/init.cmake]: OK, thank you. I will fix it.

File: build/cmake/init.cmake (hunk 2 above)
[the remark this answers is not in the document]
Comment (Deamhan) [reply build/cmake/init.cmake]: Done

**Comment by Deamhan** [comment]
About Clang: optimizations and no-rtti must work, about pedantic flags and static linking I'm not sure (I have no experience using it). Did you enable this options when you test it on clang? Some note about optimization flags: in my practice I have seen cases when using -O3 (GCC) caused generating of damaged code, what about Clang? Is using -O3 safe enough?

File: build/cmake/init.cmake (hunk 1 above)
Comment (Deamhan) [reply build/cmake/init.cmake]: Done.

**Comment by Deamhan** [comment]
@MaartenBent, about clang on windows: are you tested it as llvm toolchain for MSVS?

**Review by MaartenBent — APPROVED** [review]

**Comment by vadz** [comment]
Thanks!

**Comment by MaartenBent** [comment]
> are you tested it as llvm toolchain for MSVS?

I use LLVM Clang, part of MSYS2/MinGW64. Not the one available via VS Installer.

I don't  know about O2/O3, CMake uses O3 as default in `Release` configuration, and 02 in `RelWithDebInfo`. I have not encountered issues with both.

**Pull request closed without merging**